### PR TITLE
Refatoring api

### PR DESCRIPTION
**Pull request by K03Rat**

### CODE REVIEW

File: webserver/repositories/userRepository.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    def add(self, user: UserModel):
+        """
+        Add a new user to the database.
+
+        Args:
+            user (UserModel): The user model to be added to the database.
+        """
+        self.session.add(user)
+        self.session.flush()
```
Comment (jmccrae) [inline webserver/repositories/userRepository.py:25]: no `session.commit()`?

File: webserver/repositories/userRepository.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        """
+        Retrieve a user by email from the database.
+
+        Args:
+            email (str): The email of the user to be retrieved.
+
+        Returns:
+            UserModel: The user model if found, otherwise None.
+        """
+        print(email)
```
Comment (jmccrae) [inline webserver/repositories/userRepository.py]: Avoid leaving println debugging in final code. Remove or change to logging

File: webserver/utilities.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    doc = docx.Document(uploaded_file)
+    text = [para.text for para in doc.paragraphs]
+    return '\n'.join(text)
+
+def serialise(model):
+    columns = [c.key for c in class_mapper(model.__class__).columns]
+    return dict((c, getattr(model, c)) for c in columns)
+
+def serialise_data_model(model):
+    return {k: v for k, v in model.__dict__.items() if not k.startswith("_")}
```
Comment (jmccrae) [inline webserver/utilities.py:18]: Duplicate of method in `posService.py`?

File: webserver/utilities.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+def process_txt_file(uploaded_file):
+    content = uploaded_file.read()
+    return content.decode("utf-8")
+
+def process_docx_file(uploaded_file):
+    doc = docx.Document(uploaded_file)
+    text = [para.text for para in doc.paragraphs]
+    return '\n'.join(text)
+
+def serialise(model):
```
Comment (jmccrae) [inline webserver/utilities.py]: This method's name is to general and does not do what it says (it doesn't write anything to disk)

**Review by jmccrae — CHANGES_REQUESTED** [review]

**Comment by AdeDoyle** [comment]
`docker compose build` breaks with error: `ModuleNotFoundError: No module named 'services.fileService'` @banerjeeshubhanker

**Pull request merged**